src/common/src/vnode_mapping/vnode_placement.rs

```diff
@@ -49,14 +49,16 @@ pub fn place_vnode(
     // evenly among workers.
     let mut selected_pu_ids = Vec::new();
     while !new_pus.is_empty() {
-        new_pus.drain_filter(|ps| {
-            if let Some(p) = ps.next() {
-                selected_pu_ids.push(p.id);
-                false
-            } else {
-                true
-            }
-        });
+        new_pus
+            .extract_if(|ps| {
+                if let Some(p) = ps.next() {
+                    selected_pu_ids.push(p.id);
+                    false
+                } else {
+                    true
+                }
+            })
+            .for_each(drop);
```

Comment:
Should we prefer `retain` in this case?

> If the returned `ExtractIf` is not exhausted, e.g. because it is dropped without iterating or the iteration short-circuits, then the remaining elements will be retained. Use `[retain](https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#method.retain)` with a negated predicate if you do not need the returned iterator.

Reply:
It's for `Vec`, code here is `LinkedList`. For `LinkedList`:

> If the returned ExtractIf is not exhausted, e.g. because it is dropped without iterating or the iteration short-circuits, then the remaining elements will be retained. Use extract_if().for_each(drop) if you do not need the returned iterator.